scapy/layers/tuntap.py

```diff
@@ -188,8 +204,38 @@ def __init__(self, iface=None, mode_tun=None, default_read_size=MTU,
                 ifrn_name=self.iface,
                 ifru_flags=flags))
 
+            sock = open(devname, "r+b", buffering=0)
             ioctl(sock, LINUX_TUNSETIFF, tsetiff)
 
+        elif BSD:
+            if not (self.iface.startswith(b"tap") or
+                    self.iface.startswith(b"tun") or
+                    self.iface.startswith(b"utun")):
+                raise ValueError("Interface names must start with `tun`, `utun`, or "
+                                 "`tap` on BSD and Darwin")
+
+            if (self.iface.startswith(b"utun")):
+                self.kernel_packet_class = MacOSUtunPacketInfo
+                self.mtu_overhead = 4
+
+                match = re.match(r"([a-z]+)([0-9]+)", self.iface.decode("utf-8"), re.I)
```

Comment:
Don't need to import `re` here or decode bytestrings - just strip the prefix (`b'utun'`) from the string:

```python
# self.iface = b'utun0'
interface_num = int(self.iface[4:])
```